test/e2e/transformations/xmltojson/main.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package xmltmtojson
+
+import (
+	"context"
+	"net/url"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	. "github.com/onsi/ginkgo/v2" //nolint:stylecheck
+	. "github.com/onsi/gomega"    //nolint:stylecheck
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+
+	"github.com/triggermesh/triggermesh/test/e2e/framework"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/apps"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/bridges"
+	e2ece "github.com/triggermesh/triggermesh/test/e2e/framework/cloudevents"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/ducktypes"
+)
+
+var transAPIVersion = schema.GroupVersion{
+	Group:   "flow.triggermesh.io",
+	Version: "v1alpha1",
+}
+
+const (
+	transformationKind     = "XMLToJSONTransformation"
+	transformationResource = "xmltojsontransformations"
+
+	expectedResponseEvent = "{\"string\":\"\\u003cnote\\u003e\\u003cto\\u003eTove\\u003c/to\\u003e\\u003cfrom\\u003eJani\\u003c/from\\u003e\\u003cheading\\u003eReminder\\u003c/heading\\u003e\\u003cbody\\u003eDont forget me this weekend\\u003c/body\\u003e\\u003c/note\\u003e\"}"
```

Comment:
This doesn't seem right. The transformation is supposed to convert structured XML data into structured JSON data, not stringify an entire XML and wrap it into a JSON.

Reply:
I think I know what happened.

`e.SetData()` assumes it receives structured data, and it tries to encode it to the given content type ([Godoc](https://pkg.go.dev/github.com/cloudevents/sdk-go/v2@v2.8.0/event#Event.SetData)).
However, you're giving it a string of pre-encoded data, so you don't want that behavior.

Try passing that data as `[]byte` instead of a string, it should help. (We might have to indicate that it's not base64 too, not 100% sure)

Reply:
Looks like passing the data as `[]byte` is working: https://go.dev/play/p/Dp9U_dJGiyJ

Reply:
I dont understand.. im not passing a string here I am passing in `readBuf` that is of type []byte


https://github.com/triggermesh/triggermesh/blob/d5d48a14c953d7ef1b218a42e3a519c091e7f767/pkg/flow/adapter/xmltojsontransformation/adapter.go#L97

https://github.com/triggermesh/triggermesh/blob/d5d48a14c953d7ef1b218a42e3a519c091e7f767/pkg/flow/adapter/xmltojsontransformation/adapter.go#L108

Reply:
I'm talking about the test and only the test.
Look at the `expectedResponseEvent`, I do hope you can see it's not the JSON representation of your XML, and the reason for that is the way you crafted your "hello" XML event.

Reply:
Thank you again!